app/src/main/assets/native-10.7/ExoPlayerPlugin.js

```diff
@@ -0,0 +1,389 @@
+    duration(val) {
+        return null;
+    }
+
+    destroy() {
+        this._nativePlayer.destroyPlayer();
+    }
+
+    pause() {
+        this._paused = true;
+        this._nativePlayer.pausePlayer();
+    }
+
+    unpause() {
+        this._paused = false;
+        this._nativePlayer.resumePlayer();
+    }
+
+    paused() {
+        return this._paused;
+    }
+
+    async stop(destroyPlayer) {
+        this._nativePlayer.stopPlayer();
+
+        if (destroyPlayer) {
+            this.destroy();
+        }
+    }
+
+    /**
+     * Get or set volume percentage as as string
+     */
+    volume(volume) {
+        if (volume !== undefined) {
+            let volumeInt = parseInt(volume);
+            this._nativePlayer.setVolume(volumeInt);
+        }
+        return null;
+    }
+
+    setMute(mute) {
+        // Assume 30% as default when unmuting
+        this._nativePlayer.setVolume(mute ? 0 : 30);
+    }
+
+    isMuted() {
+        return false;
+    }
+
+    seekable() {
+        return true;
+    }
+
+    seek(ticks) {
+        this._nativePlayer.seek(ticks);
+    }
+
+    currentTime(ms) {
+        if (ms !== undefined) {
+            this._nativePlayer.seekMs(ms);
+        }
+        return this._currentTime;
+    }
+
+    setSubtitleStreamIndex(index) {
+        this.subtitleStreamIndex = index;
+    }
+
+    canSetAudioStreamIndex() {
+        return false;
+    }
+
+    setAudioStreamIndex(index) {
+        this.audioStreamIndex = index;
+    }
+
+    async changeSubtitleStream(index) {
+        // detach from the main ui thread
+        var innerIndex = Number(index);
+        this.playbackManager.setSubtitleStreamIndex(innerIndex);
+        this.subtitleStreamIndex = innerIndex;
+    }
+
+    async changeAudioStream(index) {
+        var innerIndex = Number(index);
+        this.playbackManager.setAudioStreamIndex(innerIndex);
+        this.audioStreamIndex = innerIndex;
+    }
+
+    prepareAudioTracksCapabilities(options) {
+        var directPlayProfiles = this.cachedDeviceProfile.DirectPlayProfiles;
+        var container = options.mediaSource.Container;
+
+        options.mediaSource.MediaStreams.forEach(function (track) {
+            if (track.Type === "Audio") {
+                var codec = (track.Codec || '').toLowerCase();
+
+                track.supportsDirectPlay = directPlayProfiles.filter(function (profile) {
+                    return profile.Container === container && profile.Type === 'Video' && profile.AudioCodec.indexOf(codec) !== -1;
+                }).length > 0;
+            }
+        });
+    }
+
+    async notifyStopped() {
+        let stopInfo = {
+            src: this._currentSrc
+        };
+
+        this.events.trigger(this, 'stopped', [stopInfo]);
+        this._currentSrc = this._currentTime = null;
+    }
+
+    async getDeviceProfile() {
+        // using native player implementations, check if item can be played
+        // also check if direct play is supported, as audio is supported
```

Comment:
While we're at it, do you want to update this comment to something meaningful?

Reply:
I just removed it